docs/cookbook/create-basic-grid.md

```diff
@@ -0,0 +1,270 @@
+    }
+```
+
+A most interesting method is `createAndGetDataSource` that should be in the same `ArticleGridFactory` and will look like this
+```php
+    /**
+     * @return \Shopsys\FrameworkBundle\Component\Grid\DataSourceInterface
+     */
+    protected function createAndGetDataSource(): DataSourceInterface
+    {
+        $queryBuilder = $this->entityManager->createQueryBuilder();
+
+        $queryBuilder->select('a')
+            ->from(Article::class, 'a');
+
+        return new QueryBuilderDataSource($queryBuilder, 'a.id');
+    }
+```
+
+## Present data with Grid
+We prepared our grid, but for now, it is not rendered anywhere and does not contain any columns.
+We change this now.
+
+First, we add columns we want to see into `create` method of `ArticleGridFactory`
+
+```diff
+    public function create(): Grid
+    {
+        $grid = $this->gridFactory->create('articleGrid', $this->createAndGetDataSource());
+
++       $grid->addColumn('id', 'a.id', t('Id'));
++       $grid->addColumn('name', 'a.name', t('Title'));
++       $grid->addColumn('createdAt', 'a.createdAt', t('Created'));
+
+        return $grid;
+    }
+```
+
+*Note: In the example above are column names translated. Do not forget to [dump translations](/docs/introduction/console-commands-for-application-management-phing-targets.md#dump-translations).*
+
+Grid is ready to show all the articles from the database.
+And we just need to render grid itself in a controller.
+For this, we reuse the one created in [Adding a New Administration Page](/docs/cookbook/adding-a-new-administration-page.md) cookbook and add new action method for presenting articles.
+Do not forget to create a twig template for the new action.
+
+We need to inject (pass through constructor) `ArticleGridFactory` created earlier and pass grid view to the template.
+
+```php
+class DashboardController extends AdminBaseController
+{
+    ...
+
+    /**
+     * @Route("/dashboard/articles/")
+     */
+    public function articlesAction()
+    {
+        $grid = $this->awesomeGridFactory->create();
+
+        return $this->render('@ShopsysShop/Admin/Content/Dashboard/articles.html.twig', [
+            'gridView' => $grid->createView(),
+        ]);
+    }
+```
+
+In the template, we render the whole grid just easy by calling `{{ gridView.render() }}` and grid will start do its work.
+
+![Display Data with Grid](img/grid-display-data.png)
+
+## Modify the basic grid appearance
+As you probably noticed, dates in the third column are not printed much friendly.
+To adjust appearance, we just need to extend the default grid template and modify it accordingly.
+
+We create the new twig template `articleGrid.html.twig` in `src/Shopsys/ShopBundle/Resources/views/Admin/Content`
+The template have to extend `@ShopsysFramework/Admin/Grid/Grid.html.twig` and override block `grid_value_cell_id_createdAt` <!--- TODO add after page will exist ([you can read more about blocks here](/docs/administration/grid-rendering.md)) -->
```

Comment:
oh, I wanted to suggest you adding some explanation on how this works (something like `grid_value_cell_id_<columnId> where columnID is set as a 1st parameter in 
Grid::addColumn() method`) but ok, the TODO is fine for me now :grin: